examples/nima/fault-tolerance/src/main/java/io/helidon/examples/nima/faulttolerance/PlatformExecutorProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.examples.nima.faulttolerance;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.helidon.common.LazyValue;
+
+import jakarta.inject.Named;
+import jakarta.inject.Provider;
+import jakarta.inject.Singleton;
+
+/**
+ * This service will be part of Níma on Pico module.
+ * It may use pico to get config sources exposed through pico.
+ */
+@Singleton
+@Named("platform-executor")
+class PlatformExecutorProvider implements Provider<ExecutorService> {
+    private static final AtomicInteger COUNTER = new AtomicInteger();
+
+    private static final LazyValue<ExecutorService> EXECUTOR_SERVICE = LazyValue.create(() -> {
+        return Executors.newFixedThreadPool(10,
+                                            r -> new Thread(r, "platform-" + COUNTER.getAndIncrement()));
+    });
+
+    @Override
+    public ExecutorService get() {
```

Comment:
this will make it into dependent scope - ie., each inject point will get a new one.  Is that what you want really?

I think a more flexible design will allow this to be an `InjectionPointProvider`, then optionally look at the name qualifier, and use a named thread pool (optionally backed by some configuration).

Reply:
That is why I am using a constant with a single value. This is an example to create a single named instance, not to handle unknown list of named instances.
Nevertheless it would be good if there was some easy way to mark that the return value should be treated as a singleton